source/FormationPattern.h

```diff
@@ -67,14 +97,111 @@ class FormationPattern {
 	void SetName(const std::string &name);
 
 	// Get an iterator to iterate over the formation positions in this pattern.
-	PositionIterator begin() const;
+	PositionIterator begin(double diameterToPx, double widthToPx, double heightToPx,
+		double centerBodyRadius, unsigned int shipsToPlace = 0) const;
+
+	// Information about allowed rotating and mirroring that still results in the same formation.
+	int Rotatable() const;
+	bool FlippableY() const;
+	bool FlippableX() const;
+
+private:
+	// Retrieve properties like number of lines and arcs, number of repeat sections and number of positions.
+	// Private, because we hide those properties and just provide a position iterator instead.
+	unsigned int Lines() const;
+	// Number of repeat sections on the current line.
+	unsigned int Repeats(unsigned int lineNr) const;
+	// Number of positions on the current repeat section of the active line or arc.
+	unsigned int Positions(unsigned int ring, unsigned int lineNr, unsigned int repeatNr) const;
+	// Tells if the current line or arc is centered.
+	bool IsCentered(unsigned int lineNr) const;
+
+	// Calculate a position based on the current ring, line/arc, repeat-section and position on the line-repeat-section.
+	Point Position(unsigned int ring, unsigned int lineNr, unsigned int repeatNr,
+		unsigned int lineRepeatPosition, double diameterToPx, double widthToPx, double heightToPx) const;
+
+private:
+	class MultiAxisPoint {
```

Comment:
Comment what this class is?

Reply:
I'll add a comment on what this class does. (It is actually part of #4471 where we can also specify formation patterns based on the sizes of the ships that are participating.)